src/test/groovy/edu/ie3/datamodel/io/source/csv/CsvDataSourceTest.groovy

```diff
@@ -0,0 +1,351 @@
+/*
+ * © 2020. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+package edu.ie3.datamodel.io.source.csv
+
+import edu.ie3.datamodel.io.FileNamingStrategy
+import edu.ie3.datamodel.models.UniqueEntity
+import edu.ie3.datamodel.models.input.NodeInput
+import edu.ie3.datamodel.models.input.OperatorInput
+import edu.ie3.test.common.SystemParticipantTestData as sptd
+import edu.ie3.test.common.GridTestData as gtd
+import spock.lang.Shared
+import spock.lang.Specification
+
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.atomic.LongAdder
+import java.util.stream.Collectors
+
+
+class CsvDataSourceTest extends Specification {
+
+	// Using a groovy bug to gain access to private methods in superclass:
```

Comment:
![really](https://media.giphy.com/media/31TsJOgYOvIWWYMEaR/source.gif)
Really, groovy?

Reply:
:D funny right? - but more important: thanks to your comment, I finally learned how to include .gifs in github comments 😏